lib/puppet/type/cisco_interface.rb

```diff
@@ -698,6 +724,206 @@ def is_to_s(value)
     end
   end # property stp_vlan_port_priority
 
+  ###########################
+  # private vlan attributes #
+  ###########################
+
+  newproperty(:switchport_mode_private_vlan_host) do
+    desc 'Switchport private host mode of the interface.'
+
+    newvalues(
+      :host,
+      :promiscuous,
+      :disabled)
+  end # property switchport_mode_private_vlan_host
+
+  newproperty(:switchport_mode_private_vlan_host_association, array_matching: :all) do
+    format = '["primary_vlan", "secondary_vlan"]'
+    desc "An array of #{format} pairs. "\
+         "Valid values match format #{format}. "\
+         'primary_vlan and secondary_vlan are integers.'
+    match_error = "must be of format #{format}. "\
+                  'primary_vlan and secondary_vlan must be specified as integers.'
+
+    validate do |value|
```

Comment:
We should support an array of strings or an array of integers in this case.  Users should not have to enter strings.